### PR TITLE
feat: Add tokenless mutation and field in owner

### PR DESCRIPTION
**Pull request by RulaKhaled**
### Purpose/Motivation
What is the feature? Why is this being done?
The user will set tokenless to either true or false. We need tokens_required to reflect the current status of tokenless within the organization, and we need a mutation to set tokenless to either required or not

this PR depends on:

- [x]  [Add new tokens_required field for owners table #2297](https://github.com/codecov/engineering-team/issues/2297)

### Links to relevant tickets
[[API] Expose new tokens_required field via graphql](https://github.com/codecov/engineering-team/issues/2301)
[[API] Add new require token mutation](https://github.com/codecov/engineering-team/issues/2516)

### What does this PR do?
Include a brief description of the changes in this PR. Bullet points are your friend.
- mutation
- new field under owner query 
- tests 

### Notes to Reviewer
Anything to note to the team? Any tips on how to review, or where to start?

### Legal Boilerplate

Look, I get it. The entity doing business as "Sentry" was incorporated in the State of Delaware in 2015 as Functional Software, Inc. In 2022 this entity acquired Codecov and as result Sentry is going to need some rights from me in order to utilize my contributions in this PR. So here's the deal: I retain all rights, title and interest in and to my contributions, and by keeping this boilerplate intact I confirm that Sentry can use, modify, copy, and redistribute my contributions, under Sentry's choice of terms.

### CODE REVIEW

File: graphql_api/types/owner/owner.graphql [hunk 1]
```diff
@@ -37,4 +37,5 @@ type Owner {
   ): RepositoryConnection! @cost(complexity: 25, multipliers: ["first", "last"])
   username: String
   yaml: String
+  tokensRequierd: Boolean
```
Comment (adrian-codecov) [inline graphql_api/types/owner/owner.graphql:40]: Typo

File: codecov_auth/commands/owner/interactors/set_tokens_required.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+            org_username=input.get("org_username"),
+        )
+
+        owner_obj = Owner.objects.filter(
+            username=typed_input.org_username, service=self.service
+        ).first()
+
+        self.validate(owner_obj)
+
+        owner_obj.tokens_required = typed_input.tokens_required
```
Comment (adrian-codecov) [inline codecov_auth/commands/owner/interactors/set_tokens_required.py:40]: Thoughts on this being token_required? And even to be more explicit, `upload_token` required? 

We use the github/providers token for other things, so differentiating this as the "codecov_token" or "upload_token" would make more sense. I also don't know if this exclusively a coverage token or if it also applies to bundle_analysis/test_results, so I'd confirm what's the expected behavior, and if it is exclusively for coverage (and we don't intend to do it for BA/TR), then I'd call it `coverage_token` or `upload_coverage_token` or something like that

Comment (RulaKhaled) [reply codecov_auth/commands/owner/interactors/set_tokens_required.py:40]: This is just a draft PR not the final code, just put `tokens_required` as it's the suggested name by Trent for Nora, so  i will change the name according to what it's called in the DB to avoid any confusion

Comment (adrian-codecov) [reply codecov_auth/commands/owner/interactors/set_tokens_required.py:40]: Sounds good! 👌

File: graphql_api/types/owner/owner.graphql (hunk 1 above)
Comment (RulaKhaled) [reply graphql_api/types/owner/owner.graphql:40]: yep, when user is not part of the org (decorator rules)

File: codecov_auth/commands/owner/interactors/set_tokens_required.py (hunk 2 above)
Comment (RulaKhaled) [reply codecov_auth/commands/owner/interactors/set_tokens_required.py]: changed to upload_token_required!

File: graphql_api/types/owner/owner.graphql (hunk 1 above)
Comment (RulaKhaled) [reply graphql_api/types/owner/owner.graphql]: Fixed 👍

File: codecov_auth/commands/owner/interactors/set_upload_token_required.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+class SetUploadTokenRequiredInteractor(BaseInteractor):
+    def validate(self, owner_obj):
+        if not self.current_user.is_authenticated:
+            raise Unauthenticated()
+        if not owner_obj:
+            raise ValidationError("Owner not found")
+        if not current_user_part_of_org(self.current_owner, owner_obj):
+            raise Unauthorized()
+        if not owner_obj.is_admin(self.current_owner):
+            raise Unauthorized("Admin authorization required")
```
Comment (adrian-codecov) [inline codecov_auth/commands/owner/interactors/set_upload_token_required.py:25]: Can upload_token_required be null? If not maybe do some validation here too

File: codecov_auth/commands/owner/interactors/tests/test_set_upload_token_required.py [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+        self.execute(
+            current_user=self.current_user,
+            org_username=self.owner_with_admins.username,
+        )
+
+        self.owner_with_admins.refresh_from_db()
+        assert self.owner_with_admins.upload_token_required_for_public_repos == True
+
+    def test_set_upload_token_required_to_false(self):
```
Comment (adrian-codecov) [inline codecov_auth/commands/owner/interactors/tests/test_set_upload_token_required.py:70]: I'd add a test here for the null case based on the above

File: graphql_api/types/mutation/set_upload_token_required/set_upload_token_required.graphql [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+    UnauthenticatedError
+  | UnauthorizedError
+  | ValidationError
+
+type SetUploadTokenRequiredPayload {
+  error: SetUploadTokenRequiredError
+}
+
+input SetUploadTokenRequiredInput {
+  org_username: String!
```
Comment (adrian-codecov) [inline graphql_api/types/mutation/set_upload_token_required/set_upload_token_required.graphql:11]: Why is this snake and the below camel 👀 could we follow the camel convention pls?

File: graphql_api/types/owner/owner.graphql [hunk 6]
```diff
@@ -39,4 +39,5 @@ type Owner {
   yaml: String
   aiFeaturesEnabled: Boolean!
   aiEnabledRepos: [String]
+  uploadTokenRequired: Boolean
```
Comment (adrian-codecov) [inline graphql_api/types/owner/owner.graphql:42]: Is it intentional for this to be null?

**Review by adrian-codecov — APPROVED** [review]

File: graphql_api/types/owner/owner.graphql (hunk 6 above)
Comment (RulaKhaled) [reply graphql_api/types/owner/owner.graphql:42]: Intertactor should always return boolean, but the decorator `@require_part_of_org` should return null if user is not part of the org!

File: graphql_api/types/mutation/set_upload_token_required/set_upload_token_required.graphql (hunk 5 above)
Comment (RulaKhaled) [reply graphql_api/types/mutation/set_upload_token_required/set_upload_token_required.graphql:11]: Great catch

File: codecov_auth/commands/owner/interactors/set_upload_token_required.py (hunk 3 above)
Comment (RulaKhaled) [reply codecov_auth/commands/owner/interactors/set_upload_token_required.py:25]: on it
